x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/QueryBuilder.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.parser;
+
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.Literal;
+import org.elasticsearch.xpack.ql.expression.UnresolvedAttribute;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.And;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.Equals;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+public class QueryBuilder extends ExpressionBuilder {
+
+    @Override
+    public Expression visitEventQuery(EqlBaseParser.EventQueryContext ctx) {
```

Comment:
The query should map to a Filter, that is a `LogicalPlan` (action) not an `Expression` (properties):
https://github.com/costin/elasticsearch/commit/5237f2b0c6e0f27d473362e742defe506569d4e5#diff-74e29c0f2741ee97378ecd3b111c6e8cR20

or 

https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/LogicalPlanBuilder.java#L125

Reply:
resolved in d976f1e9ec1c6cb4eaa712bde3352142500bb53f